Doc/c-api/module.rst

```diff
@@ -451,8 +451,9 @@ state:
 
    On success, return ``0``. On error, raise an exception and return ``-1``.
```

Comment:
Looks like the second sentence here is redundant.

Reply:
> [...] or leave that for the user to check.

I'm -1 for that option; the reference should be succinct and exact (see [Diátaxis](https://diataxis.fr/reference/)).

I also think we're better off with removing the _"On error..." sentence above; we've covered the error case in your new and improved text below.

Reply:
Marking as resolved; follow-up in https://github.com/python/cpython/pull/101957#discussion_r1175816245